packages/terra-table/src/SelectableUtils.js

```diff
@@ -16,21 +29,76 @@ const initialSingleSelectRowIndex = (rows) => {
 };
 
 /**
-* Returns if the selected row index is already selected for SingleSelectableRows.
-*/
+ * Returns the indexes of the selected rows for MultiSelectableRows.
+ * To be used in the constructor, to set initial state.
+ */
+const initialMultiSelectRowIndexes = (rows, maxSelectionCount) => {
+  if (!rows || !rows.length) {
+    return [];
+  }
+  // Find the rows which are selected and are selectable
+  const selectedIndexes = [];
+  const validatedMaxSelectionCount = validatedMaxCount(rows, maxSelectionCount);
+  for (let i = 0; i < rows.length; i += 1) {
+    if (selectedIndexes.length >= validatedMaxSelectionCount) {
```

Comment:
could this get moved into the isSelected if statement to after you push on the selected index,

this way it wouldn't get checked every loop and would still be able to break out early after finding all the indexes

Reply:
Moved this check to inside the if statement after the push